### PR TITLE
CFE-2886: Changed static functions in verify_environments to use Attributes *

### PR DESCRIPTION
**Pull request by olehermanse**
Reported by LGTM:
https://lgtm.com/rules/2163210742/

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom jenkins

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1250/)

https://ci.cfengine.com/job/pr-pipeline/1250/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/1254/)

https://ci.cfengine.com/job/pr-pipeline/1254/

File: cf-agent/verify_environments.c [hunk 1]
```diff
@@ -158,35 +158,37 @@ PromiseResult VerifyEnvironmentsPromise(EvalContext *ctx, const Promise *pp)
 
 /*****************************************************************************/
 
-static int EnvironmentsSanityChecks(Attributes a, const Promise *pp)
+static int EnvironmentsSanityChecks(const Attributes *a, const Promise *pp)
 {
-    if (a.env.spec)
+    assert(a != NULL);
+    const Environments env = a->env;
```
Comment (vpodzime) [inline cf-agent/verify_environments.c]: Do we want to create this copy?

File: cf-agent/verify_environments.c [hunk 2]
```diff
@@ -199,12 +201,13 @@ static int EnvironmentsSanityChecks(Attributes a, const Promise *pp)
...
 
-static PromiseResult VerifyEnvironments(EvalContext *ctx, Attributes a, const Promise *pp)
+static PromiseResult VerifyEnvironments(EvalContext *ctx, const Attributes *a, const Promise *pp)
 {
+    assert(a != NULL);
     char hyper_uri[CF_MAXVARSIZE];
     enum cfhypervisors envtype = cfv_none;
-
-    switch (Str2Hypervisors(a.env.type))
+    Environments env= a->env;
```
Comment (vpodzime) [inline cf-agent/verify_environments.c]: Same question here. Also, there's a space missing before the `=`.

File: cf-agent/verify_environments.c (hunk 1 above)
Comment (olehermanse) [reply cf-agent/verify_environments.c]: Environments struct is relatively small (10 ints/pointers). I don't think this is a big deal, but using a pointer or copying just the parts needed are just as valid, imo.

**Pull request merged**